csm/core/services/stats.py

```diff
@@ -160,6 +176,37 @@ async def get_metrics(self, stats_id, metrics_list, from_t, to_t, interval,
         Log.debug(f"Stats Request Output: {output}")
         return output
 
+    def _convertor(self, message):
+        converted_message = self.convertor.convert_data(message)
+        return converted_message
+
+    def _stats_callback(self, message):
+        # TODO: call self._convertor(message) for conversion of metric
+        #converted_message = self._convertor(message)
+        #StatsAppService.BUFFER.append(converted_message)
+        StatsAppService.BUFFER.append(message)
+
+    def _parse_metrics(self, messages):
+        metrics = ""
+        for metric in StatsAppService.BUFFER:
+            # Add new line while parsing if metrics are not read from a static file
+            metrics += str(metric)
+        return web.Response(text=metrics)
+
+    async def get_perf_metrics(self):
+        """ get performace stat metrics"""
+        StatsAppService.BUFFER = []
+        # Read from a static file and publish to msassage bus
+        with open("/opt/seagate/cortx/csm/templates/metrics.txt", "r") as fp:
```

Comment:
Is this file checked in? Have you made sure this file is available in the same location?

Reply:
yes. this file is already present.